app/src/common/shared/com/igalia/wolvic/ui/adapters/WebApp.java

```diff
@@ -0,0 +1,163 @@
+package com.igalia.wolvic.ui.adapters;
+
+import android.graphics.Color;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.igalia.wolvic.utils.StringUtils;
+import com.igalia.wolvic.utils.SystemUtils;
+
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Objects;
+import java.util.OptionalInt;
+
+/**
+ * Parses and encapsulates (a subset of) the fields in a Web App manifest.
+ * See https://www.w3.org/TR/appmanifest/ for reference.
+ */
+public class WebApp {
+
+    protected final String LOGTAG = SystemUtils.createLogtag(this.getClass());
+
+    @NonNull
+    private String mIdentity;
+    @NonNull
+    private String mName;
+    @NonNull
+    private String mShortName;
+    @NonNull
+    private String mScope;
+    @NonNull
+    private String mStartUrl;
+
+    private int mThemeColor;
+    private int mBackgroundColor;
+
+    private OptionalInt mHashCode = OptionalInt.empty();
+
+    // TODO icons, shortcuts, languages
+
+    /**
+     * Will throw {@link java.io.IOException} if there is an error parsing the manifest.
```

Comment:
I think it's better to use the @throws (or @exception) tag here for the documentation

Reply:
Done.